pylint/checkers/refactoring.py

```diff
@@ -658,12 +663,39 @@ def _find_lower_upper_bounds(comparison_node, lower_bounds, upper_bounds):
             self.add_message('chained-comparison',
                              node=node)
 
+    def _check_possible_any_all_usage(self, boolop_node):
+        if boolop_node.op not in ('and', 'or'):
+            return
+
+        def _get_operands(bool_op, operator):
+            values = []
+            if bool_op.op != operator:
+                return values
+
+            for value in bool_op.values:
+                if isinstance(value, astroid.Name):
+                    values.append(value.name)
+                elif isinstance(value, astroid.Call):
```

Comment:
We should limit this check just to names. Due to the short-circuiting behaviour of the boolean expressions, we can move performance critical checks at the end of the expression, making them trigger only if the previous conditions held true:
```
if some_value and some_other_value and do_expensive_check():
   pass
```

Moving this to `all([some_value, some_other_value, do_expensive_check()])` means we now lose that performance improvement and we evaluate `do_expensive_check` all the time.

Reply:
That's definitely not true, as the values have to be stored in a sequence before `any` could get to them, resulting in the calls to be evaluated:
```
In [1]: def test(): print('evaluated')

In [2]: any((1, 2, test()))
evaluated
Out[2]: True
```

Reply:
ohh I didn't catch that. I will then have this check only for names.